src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -42,7 +44,7 @@ public sealed partial class Pipe
         private readonly PipeScheduler _readerScheduler;
         private readonly PipeScheduler _writerScheduler;
 
-        private readonly BufferSegment[] _bufferSegmentPool;
+        private readonly Stack<BufferSegment> _bufferSegmentPool;
```

Comment:
What's the reason for using the stack here? It's another allocation per pipe and it's 3 fields instead of 2 that we had before.

Reply:
We can live with another field. Also we saved on one the other PR so it’s a trade

Reply:
But what are we gaining by spending a field and allocation?

Reply:
Clearer code and a way to resize the initial buffer up to the max without writing that code manually.

Reply:
It's literally 3 lines of code:

https://github.com/dotnet/corefx/blob/2a2d34df7db5d2617d12f55fe5ad8fd208117049/src/System.Collections/src/System/Collections/Generic/Stack.cs#L286-L289